### PR TITLE
Cvsl 1626 cvl sar filter returned data

### PR DESCRIPTION
**Pull request by nishanthmahasamudram**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/model/Views.kt [hunk 1]
```diff
@@ -0,0 +1,5 @@
+package uk.gov.justice.digital.hmpps.createandvaryalicenceapi.model
+
+class Views {
+  interface PublicSar
```
Comment (andrewrlee) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/model/Views.kt:4]: can we just call the `SubjectAccessRequest` ?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/model/Licence.kt [hunk 2]
```diff
@@ -37,24 +38,29 @@ sealed interface Licence {
   val id: Long
```
Comment (andrewrlee) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/model/Licence.kt:38]: if we're exposing licence id on events we should probably expose it here too

Comment (andrewrlee) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/model/Licence.kt:38]: should probably expose kind too

Comment (nishanthmahasamudram) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/model/Licence.kt:38]: we are already exposing kind.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/model/AdditionalCondition.kt [hunk 3]
```diff
@@ -8,29 +9,43 @@ data class AdditionalCondition(
...
 
   @Schema(description = "Sequence of this additional condition within the additional conditions", example = "1")
   val sequence: Int? = null,
 
-  @Schema(description = "The textual value for this additional condition", example = "You must not enter the location [DESCRIPTION]")
+  @Schema(
+    description = "The textual value for this additional condition",
+    example = "You must not enter the location [DESCRIPTION]",
+  )
+  @get:JsonView(Views.PublicSar::class)
```
Comment (andrewrlee) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/model/AdditionalCondition.kt:30]: think we're ok without this - as it should be covered in the expanded text

File: src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/resource/publicApi/SubjectAccessRequestController.kt [hunk 4]
```diff
@@ -27,6 +29,7 @@ import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.service.publicApi.S
 @RequestMapping("/public", produces = [MediaType.APPLICATION_JSON_VALUE])
```
Comment (andrewrlee) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/resource/publicApi/SubjectAccessRequestController.kt:29]: Just noticed this, but it shouldnt be on the /public path

File: src/test/resources/test_data/sar_content/hardstopLicence.json [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+            "code": "notBreakLaw",
+            "text": "Do not break any law"
+          },
+          {
+            "code": "attendMeetings",
+            "text": "Attend meetings"
+          }
+        ],
+        "standardPssConditions": [],
+        "additionalLicenceConditions": [],
```
Comment (andrewrlee) [inline src/test/resources/test_data/sar_content/hardstopLicence.json:41]: shouldn't this have an additional condition fed through? its added by default through test data

File: src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/resource/publicApi/SubjectAccessRequestController.kt (hunk 4 above)
Comment (nishanthmahasamudram) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/resource/publicApi/SubjectAccessRequestController.kt:29]: private?

File: src/test/resources/test_data/sar_content/hardstopLicence.json (hunk 5 above)
Comment (nishanthmahasamudram) [reply src/test/resources/test_data/sar_content/hardstopLicence.json:41]: No I am passing emptyMap() for conditionSubmissionStatus

Comment (nishanthmahasamudram) [reply src/test/resources/test_data/sar_content/hardstopLicence.json]: Updated the test case to fetch additional conditions

File: src/test/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/TestData.kt [hunk 6]
```diff
@@ -28,7 +28,7 @@ object TestData {
     licence = licence,
     id = 2L,
     conditionSequence = 1,
-    conditionType = HARD_STOP_CONDITION.type,
+    conditionType = "AP",
```
Comment (andrewrlee) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/TestData.kt:31]: good spot!

**Pull request merged**